README.md

```diff
@@ -1,77 +1,337 @@
+representation of said component, so why not house them together in one file? It's convenient!
 
-Just spitballing, but I think it might be nice to have them write a `bin/build`
-script (in JavaScript) that makes use of browserify's Node.js API. They can then
-add `"build": "node bin/build"` to the package.json's `"scripts"` field to get a
-sense of how that all works. (One aim here is to demystify some of the command
-line magic — it's just scripts!)
+It's important to note that using JSX is _entirely_ optional. If writing HTML in your JavaScript feels like committing a
+mortal sin, it's okay to use `React.createElement()`. You'll be less productive, but at least you'll feel good about it!
 
-Make sure they understand the presets that they'll need (`react` and `es2015`
-should do it for now).
+Once we have our JSX, we'll need to perform an additional step to get this to work in our code. While this is valid JSX
+code, it's not something that browsers understand. To compile our code into something that the browsers understand, we
+use Babel that transpiles our code down to working JS that all browsers can handle. More on that later!
 
-Finally, have students write a simple component using JSX (emphasizing that it's
-really _just JavaScript_) and render it in the DOM (with the build step). So in
-other words, it would be awesome to have students set up some kind of basic
-directory structure:
+## Gotchas
+In the above code, you'll see that we're returning _one_ XML element (the `.tweet` div). JSX always has one, and _only_
+one element (that optionally has children, grandchildren, and so on). You'd think we could do something like this:
 
+```js
+return (
+  <p>I am the first paragraph</p>
+  <p>I am the second paragraph</p>
+);
 ```
-index.js
-components/
-|
-|_ HelloWorld.js
+
+But we can't! We'd have to wrap these two paragraphs in a `div` element:
+
+```js
+return (
+  <div>
+    <p>I am the first paragraph</p>
+    <p>I am the second paragraph</p>
+  </div>
+);
 ```
 
-Where in their `index.html` they're including
+The reason for this becomes abundantly clear when we take a look at the compiled output for the above code sample. This
+is the same code, but compiled down to regular JS:
 
-```html
-<script src="build/index.js"></script>
+```js
+return React.createElement(
+  "div",
+  null,
+  React.createElement(
+    "p",
+    null,
+    "I am the first paragraph"
+  ),
+  React.createElement(
+    "p",
+    null,
+    "I am the second paragraph"
+  )
+);
+```
+
+As you can see, it's returning _one_ element with its children. Not being able to just write two `<p>` elements in our
+earlier code sample now makes sense — returning two values at once in JavaScript is conceptually impossible.
+
+Another thing to note is that since we're still writing JS code, we need to avoid using keywords in our code. You might
+have noticed it already, but we're setting HTML classes using the `className` attribute (or prop, in React terms),
```

Comment:
Might not need the "but" here: "You might have noticed it already. We're setting..."